### ISSUE TITLE
Add the possibility to tag photos from Gallery

### ISSUE THREAD

**Issue by Spartachetto** [nextcloud/gallery#153]
## Feature request

**User type**: Logged-in 

**User level**: Beginners

### Description
When I upload a group of photos, I usually add tags based on the content of these.

This mean that I look at them to tag.
Up to now this works like that:

1. open the gallery looking for photos with detail "x" (e.g.: with a specific person)
2. try to remember the photos (!)
3. switch to file view
4. scan the list trying to find all the photos you identified at step 1 (!)

these steps are done **at least once for each tag**.
If there are many photos in the directory I usually repeat the process twice or more

This is boring and time consuming.
A possible solution would be to add the possibility to tag the photos in the gallery and / or in the slideshow

### Benefit / value
This feature would improve greatly the usability for the users that are tagging the photos

### Risk / caveats
It is necessary to develop an interface intuitive and NOT disturbing for the ones that do not want to set tags. This has to be integrated with #60

### Sponsorship
 
**Are you a developer willing to implement this feature?**: no

**Can you sponsor the development of this feature or do you know someone who can?**: no

**Comment by elpraga** [comment]
I was just about to ask for the same feature. I would be a great enhancement while working with photos.
In the first phase just adding tags could be supported. 

The tags might be displayed in both views on top of the photos. with a x to remove  a specific tag and + to add a new tag.

Of course, it is only a suggestion.

I'm sorry, I cannot contribute to code it, as I cannot code, and I do understand that adding new features depends on your time.

Thanks for making the Gallery app available for us!

**Comment by joephein** [comment]
I started to use tagspaces as a desktop client but it is client local only and not cloud based. It is very advanced and has multiple options to show photo's as well (grid view, ImageSwiper). However there is no good integration with the tag system in Nextcloud. Tagspaces is whether adding the tags to the file names or working with hidden .ts files. There is a conversation regarding this as well on the Nextcloud forum: https://help.nextcloud.com/t/tagspaces-for-owncloud/452 I would love to see some progress regarding tagging and filtering as well, especially in the gallery.

**Comment by elpraga** [comment]
Any progress by a chance on this issue? 

I just had an idea when I noticed the sharing button when previewing the images.

Adding another similar **tag** button that would manage the tags for that photo should be quite straightforward, am I right? 

Viewing tagged photos from the **tag** menu in Files app works nice already.

Even though being able to tag multiple photos at once would be very useful, being able to tag individual photos would already be a great step forward. 

What do you think @oparoz ?

**Comment by elpraga** [comment]
Another possibility would be to display the whole *Details* side pane from Files app when requested (in full preview and in gallery view). It already can access tags, comments, and with an application also metadata.

...I am just brainstorming here in hope it might help..

**Comment by hypervtechnics** [comment]
Any progress on this? Just searched for this feature for 30 minutes until I noticed it doesn't exist.

**Comment by oparoz** [comment]
Nothing, no. There is no GUI element in the slideshow which would make it possible to add the future atm.

**Comment by MrRusch** [comment]
**I second this feature request!** 
Displaying the entire Details pane in the Gallery app would be a great enhancement. 

The primary reason I see for this is to allow for the additions of descriptions on photos - e.g. through the Comments tab - as there really is no way to do this today AFAIK. 
Gaining access to the Meta data of photos is another pro. 
Also, the sharing function (which is currently separated) could be removed as it has its own tab in the Details pane.

**Comment by thecanfield** [comment]
I also would love to see this implemented.

**Comment by gaiz-k** [comment]
Mee too.

**Comment by MikaelMollberg** [comment]
This would be great to have

**Comment by bananajota** [comment]
Any update? it will be great improvment. THank you!

**Comment by jkingcis** [comment]
That enhacement would fit most use cases. I think it'd be ideal to have it implemented. I hope seeing this new feature soon :)

**Comment by dankasak** [comment]
Perhaps if someone can describe in detail what would be involved in implementing this feature, some volunteers might step up. I personally hate PHP as much as any decent person, but I can make stuff happen while grinding my teeth. So ... what's involved? When it's said "There is no GUI element in the slideshow which would make it possible" ... we're basically just talking about multi-select functionality in the gallery, right? Surely that can't be too much work?

**Comment by hypervtechnics** [comment]
@dankasak Afaik this is not about multi-select. It is about the possibility to add/remove tags for a picture without leaving the slideshow/fullscreen view. This allows you to go through all photos in a directory in fullscreen and directly assign tags to it. A great boost in UX would be to use more frequent used tags at the top on the list.

**Comment by aleksandrmetik** [comment]
Display current tags will also be very cool.

**Comment by freechelmi** [comment]
FYI : in NC17 you can display the details pane in the slideshow , but the details pane is not refreshed when you go forward in the slideshow :-(

**Comment by elpraga** [comment]
Yes,  that does make it slightly less cumbersome, but still quite far from a desirable workflow. 

Closing the side panel and opening it again updates it,  but still not an enjoyable experience.
